clang/include/clang/Basic/AttrDocs.td

```diff
@@ -5557,6 +5557,21 @@ See :doc:`LTOVisibility`.
   }];
 }
 
+def TypeVisibilityDocs : Documentation {
+  let Category = DocCatType;
+  let Content = [{
+The ``type_visibility`` attribute allows the ELF visibility of a type and its vague
+linkage objects (vtable, typeinfo, typeinfo name) to be controlled separately from
+the visibility of functions and data members of the type.
+
+For example, this can be used to give default visibility to the typeinfo and the vtable
+of a type while still keeping hidden visibility on its member functions and static data
+members.
+
+This attribute can only be applied to types and namespaces.
```

Comment:
Maybe we also want to document the behavior of adding both `clang::type_visibility` and `gnu::visibility` to a namespace? (I think `clang::type_visibility` overrides `gnu::visibility`)

Reply:
You seem to be right based on https://godbolt.org/z/bbs7478sv. I'll add the documentation.

Reply:
This applies to classes too: https://godbolt.org/z/673qsnxsP